fix: 툴팁/화살표 위치 조건문/계산식 수정

rlacodud opened the pull request:
- 퍼센트 수식 수정
- 툴팁 위치 조건문 수정
- progressWidth 변수명 수정
- 가격 단위 addComma 작업

rlacodud committed:
feat: 프로그레스바 1차 백업

rlacodud committed:
fix: progressbar 피드백 반영

- 퍼센트 수식 수정
- 툴팁 위치 조건문 수정
- progressWidth 변수명 수정

rlacodud committed:
feat: 툴팁/화살표 위치 조건문/계산식 수정

rlacodud committed:
chore: 고정값 수정

spjoo commented on pages/index.vue:
```diff
@@ -1,14 +1,162 @@
+// (예: 좌측에 있을 때는 진행률에 따라 좌측으로 화살표가 이동되어야 할 수 있음)
+// Tooltip의 화살표는 항상 Tooltip 영역 범위 안에 있어야 하며 좌/우측 최소 6px의 마진이 있어야 함
+// Tooltip은 Progress bar의 영역을 벗어날 수 없음
+
```
ref, reactive, onMounted, nextTick, computed import 해서 사용해야 합니다.

spjoo commented on pages/index.vue:
```diff
@@ -1,14 +1,162 @@
+        transform: translateX(-50%);
+        width: 0;
+        height: 0;
+        border-top: 3px solid transparent;
```
왜 작성한걸까요?

spjoo commented on pages/index.vue:
```diff
@@ -1,14 +1,162 @@
+    arrowLeft = 0;
+  }
+  tooltip.value.style.transform = `translateX(${tooltipLeft}px)`;
+  arrow.value.style.marginLeft = `${arrowLeft}px`;
```
가변적으로 움직이는 값을 transform을 사용하는게 좋습니다.

spjoo commented on pages/index.vue:
```diff
@@ -1,14 +1,162 @@
+});
+
+const handleTooltipPosition = async () => {
+  await nextTick(); // DOM 업데이트가 처리될 때까지 기다림
```
DOM 이 처릴될때까지 기다리는건 페이지 로드하고 한번인데, 
함수내에 있으면 실행될때 마다 기다리기 때문에 리소스 낭비가 발생하는거 같아요.

spjoo commented on pages/index.vue:
```diff
@@ -1,14 +1,162 @@
+const handleTooltipPosition = async () => {
+  await nextTick(); // DOM 업데이트가 처리될 때까지 기다림
+  
+  const progressbarWidth = progressbar.value.clientWidth;
```
함수가 실핼될때마다 계산되진 않아도 되는 값이긴 합니다.

spjoo commented on pages/index.vue:
```diff
@@ -1,14 +1,162 @@
+  const tooltipWidth = tooltip.value.clientWidth / 2;
+  const tooltipPosLeft = tooltip.value.offsetLeft;
+  // 좌측을 넘을 때
+  if (progressbarWidth < currentbarWidth + tooltipWidth) {
```
우측으로 넘을때이지 않나요?

spjoo commented on pages/index.vue:
```diff
@@ -1,14 +1,162 @@
+  // 좌측을 넘을 때
+  if (progressbarWidth < currentbarWidth + tooltipWidth) {
+    tooltipLeft = progressbarWidth - (currentbarWidth + tooltipWidth * 2);
+    arrowLeft = minMargin < progressbarWidth - currentbarWidth
```
이거 수식 케이스 안맞아요

rlacodud committed:
fix: 피드백 반영

- ref, reactive, onMounted, onUpdated, computed import
- arrow transform으로 변경
- onUpdated 시 호출로 변경
- 툴팁/화살표 위치 수식 수정
